src/main/java/io/apicurio/datamodels/core/validation/rules/other/SecurityRequirementUniquenessValidationRule.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2019 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.datamodels.core.validation.rules.other;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import io.apicurio.datamodels.core.models.common.SecurityRequirement;
+import io.apicurio.datamodels.core.validation.ValidationRule;
+import io.apicurio.datamodels.core.validation.ValidationRuleMetaData;
+import io.apicurio.datamodels.openapi.v2.models.Oas20Operation;
+
+/**
+ * @author cfoskin@redhat.com
+ */
+public class SecurityRequirementUniquenessValidationRule extends ValidationRule {
+
+    /**
+     * Constructor.
+     * 
+     * @param ruleInfo
+     */
+    public SecurityRequirementUniquenessValidationRule(ValidationRuleMetaData ruleInfo) {
+        super(ruleInfo);
+    }
+
+    /**
+     * @param currentSecurityRequirement
+     * @param securityRequirementToCompare
+     * @return if duplicate security requirements are found
+     */
+    private boolean checkForDuplicateSecurityRequirement(SecurityRequirement currentSecurityRequirement,
+            SecurityRequirement securityRequirementToCompare) {
```

Comment:
I would expect just one compare method named something like `compareTo` or `isEqual` or something like that.  Example:

```java
private boolean isEqual(SecurityRequirement req1, SecurityRequirement req2) {
  // entire algorithm that compares the two objects
}
```

Reply:
👍